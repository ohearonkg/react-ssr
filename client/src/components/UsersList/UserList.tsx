import * as React from "react";
import { connect } from "react-redux";
import { IUser } from "../../models/User";
import { ApplicationState } from "../../reducers";

interface IUserList {
  /**
   * Users to be rendered
   */
  users: IUser[];
}

const UserList = (props: IUserList) => {
  const renderUsers = () => {
    return props.users.map((user: IUser) => (
      <li key={user.id}> {user.name} </li>
    ));
  };

  return (
    <div>
      <h1> User List </h1>
      <ul> {renderUsers()} </ul>
    </div>
  );
};

const mapStateToProps = (state: ApplicationState) => {
  return {
    users: state.users.users
  };
};

export { UserList };
export default connect(mapStateToProps)(UserList);
