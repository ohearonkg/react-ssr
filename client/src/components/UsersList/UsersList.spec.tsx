import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import UserList from "./UserList";

describe("Users List", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<UserList />);
  });

  /**
   * Snapshot
   */
  it("Should match its snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
