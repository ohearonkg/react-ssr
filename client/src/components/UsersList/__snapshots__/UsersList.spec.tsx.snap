// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users List Should match its snapshot 1`] = `
<div>
  <h1>
     User List 
  </h1>
  <ul>
     
    <li
      key="1"
    >
       
      a
       
    </li>
    <li
      key="2"
    >
       
      b
       
    </li>
    <li
      key="3"
    >
       
      c
       
    </li>
     
  </ul>
</div>
`;
